src/parser/mage/fire/modules/features/HotStreakWastedCrits.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import HIT_TYPES from 'game/HIT_TYPES';
+import EnemyInstances, { encodeTargetString } from 'parser/shared/modules/EnemyInstances';
+
+const debug = false;
+
+const PROC_WINDOW_MS = 200;
+
+const HOT_STREAK_CONTRIBUTORS = [
+  SPELLS.FIREBALL.id,
+  SPELLS.PYROBLAST.id,
+  SPELLS.FIRE_BLAST.id,
+  SPELLS.SCORCH.id,
+  SPELLS.PHOENIX_FLAMES_TALENT.id,
+];
+
+class HotStreakWastedCrits extends Analyzer {
+  static dependencies = {
+    enemies: EnemyInstances,
+  };
+
+  lastCastEvent = 0;
+  wastedCrits = 0;
+  hasPyromaniacProc = false;
+
+  constructor(...args) {
+    super(...args);
+    this.hasPyromaniac = this.selectedCombatant.hasTalent(SPELLS.PYROMANIAC_TALENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this._onCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this._onDamage);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.HOT_STREAK), this.checkForPyromaniacProc);
+
+  }
+
+  //When a spell that contributes towards Hot Streak is cast, get the event info to use for excluding the cleaves from Phoenix Flames on the damage event
+  _onCast(event) {
+    if (!HOT_STREAK_CONTRIBUTORS.includes(event.ability.guid) && event.ability.guid !== SPELLS.FLAMESTRIKE.id) {
```

Comment:
Should probably only not return early on flamestrike if this.hasPyromaniac -- otherwise it's not considered a contributor; unless I misunderstand the logic here

Reply:
i dont really remember why flamestrike is in this ... it was in the old one and i carried it over anyway cause im sure there is some reason i added it here ... but i dont remember why.

Reply:
oh ok, flamestrike was here before because of the expired procs check, i need to remove that from this and add it to the expired procs event listener instead.

Reply:
I thought flamestrike was in here because of pyromaniac? FS has a chance to proc it?

Reply:
Umm, I don't think so, cause I'm pretty sure the extent of my Pyromaniac check is just whether the player instantly got a new hot streak and has nothing to do with what was cast.